frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -173,15 +175,22 @@ public WebResourceResponse shouldInterceptRequest(final WebView view, WebResourc
             }
 
             @Override
-            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+            public boolean shouldOverrideUrlLoading(WebView view,  WebResourceRequest request) {
                 // Block navigating to any external site inside the app.
                 // This is only called if the whole page is about to change. Changes inside an iframe proceed normally.
+                String url = request.getUrl().toString();
+
                 try {
-                    // Allow opening in external browser instead, for links clicked in an onramp widget.
-                    Pattern pattern = Pattern.compile("^file:///buy/.*$");
-                    if (pattern.matcher(view.getUrl()).matches()) {
-                        Util.systemOpen(getApplication(), url);
-                        return true;
+                    // Allow opening in external browser instead, for listed domains.
+                    List<Pattern> patterns = new ArrayList<>();
+                    patterns.add(Pattern.compile(".*pocketbitcoin\\.com$"));
+                    patterns.add(Pattern.compile(".*moonpay\\.com$"));
```

Comment:
These would also match "fakepocketbitcoin.com". Didn't check what getHost() contains, but maybe sth like "^pocketbitcoin\\.com$" and ".+\\.pocketbitcoin\\.com$" would be better?

I am also trying to remember the purpose of this whitelist here. `systemOpen` already has a whitelist, so I wonder if we could just skip any checks here (and in Qt's main.cpp). Wdyt?

Reply:
good point about the regex! Will fix it. Are you sure `systemOpen` has a white list? afaict this is needed because the URLs checked here do not pass through the whitelist in the backend and are directly opened (using an external browser instead of the current view, to avoid loosing the current workflow). I think we need this to support and verify links in the iframes.

Reply:
I think the reason for the above check was that rogue JS could not redirect anywhere (except on the buy page). The check you have now looks indeed redundant to the whitelist already in systeOpen. After considering it more, I think it's fine to skip any checks here.

Reply:
Nevermind, @Beerosagos made me aware that this systemOpen is not the backend's systemOpen, so this fix is appropriate!